### DISCUSSION TITLE
Memory grows forever, is it leaking?

### DISCUSSION THREAD

**Discussion by kvnlnt** [pocketbase/pocketbase#2773] [Q&A]
- pocketbase version 0.16.0

I checked my pocketbase service running on my server with `systemctl status pocketbase.service` and the Memory was at `2.5gb`. I restarted the service and it dropped to `25mb`. I can see that every time I request data from the api the memory grows and grows. I've written a `.timer` service to restart it every night but do I have to? Seems like a memory leak.

**Comment by ganigeorgiev** [comment]
I don't think there is a "memory leak". The last time I checked (https://github.com/pocketbase/pocketbase/issues/969#issuecomment-1308520247 and https://github.com/pocketbase/pocketbase/discussions/1224#discussioncomment-4367945) I wasn't able to find a unclosed file descriptors or forgotten resources.

My guess is that the reported memory is old/unreleased memory by the garbage collector kept for optimization purpose (eg. to reuse for later) since allocating new memory and releasing it always have a cost.

If you are memory constrained, you can enforce the GC to run more often using the [`GOMEMLIMIT=500MiB`](https://pkg.go.dev/runtime@master#hdr-Environment_Variables) env variable (_it is a soft-limit_).

**Reply by kvnlnt** [reply]
ok thanks @ganigeorgiev, GC related issue makes sense, I'll give the `GOMEMLIMIT` a go
